slimta/queue/__init__.py

```diff
@@ -328,9 +328,9 @@ def enqueue(self, envelope):
         results = list(zip(envelopes, ids))
         for env, id in results:
             if not isinstance(id, BaseException):
-                if self.relay:
-                    self.active_ids.add(id)
-                    self._pool_spawn('relay', self._attempt, id, env, 0)
+                if self.relay and id not in self.active_ids:
+                        self.active_ids.add(id)
+                        self._pool_spawn('relay', self._attempt, id, env, 0)
```

Comment:
Sorry one more thing, pull back the indentation one more level on these two lines and you'll be good to go!